### PR TITLE
WebexAdapter updates

### PR DESCRIPTION
**Pull request by gabog**
This update involves several tweaks and cleanup for the webex adapter (relates to issue #2542)
 

- Updated WebexAdapter to implement IBotFrameworkHttpAdapter
- Added constructor to WebexAdapter that takes IConfiguration
- Removed Identity and GetIdentityAsync from WebexClientWrapter (this is done using GetMeAsync in WebexAdapter.ProcessAsync per request.
- Removed the auto configuration logic from WebexClientWrapper (this should be done manually to avoid multiple webhook registrations, added a TODO pleacehoder in readme so we can document this before release).
- Simplified Webex sample by:
  -  Removing unused classes.
  -  Simplified startup.cs
  -  removed MicrosoftAppId and MicrosoftAppPassword

Unrelated: removed excludes for UriParametersShouldNotBeStrings and UriPropertiesShouldNotBeStrings from ruleset and updated Twilio config settings and reated files.

### CODE REVIEW

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Webex/WebexAdapter.cs [hunk 1]
```diff
@@ -7,26 +7,42 @@
...
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebexAdapter"/> class using configuration settings.
+        /// </summary>
+        /// <param name="configuration">An <see cref="IConfiguration"/> instance.</param>
+        /// <remarks>
+        /// The configuration keys are:
+        /// AccessToken: An access token for the bot.
+        /// PublicAddress: The root URL of the bot application.
+        /// Secret: The secret used to validate incoming webhooks.
+        /// Secret: A name for the webhook subscription.
```
[the remark this answers is not in the document]
Comment (gabog) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Webex/WebexAdapter.cs]: Done (will go in the next commit, thanks)

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs [hunk 2]
```diff
@@ -120,12 +120,21 @@ public static Activity RequestToActivity(HttpRequest httpRequest, string validat
         /// generated URL signature used to validate incoming requests.</param>
         /// <param name="authToken">The authentication token for the Twilio app.</param>
         /// <exception cref="AuthenticationException">Validation failed.</exception>
-        private static void ValidateRequest(HttpRequest httpRequest, Dictionary<string, string> body, string validationUrl, string authToken)
+        private static void ValidateRequest(HttpRequest httpRequest, Dictionary<string, string> body, Uri validationUrl, string authToken)
         {
             var twilioSignature = httpRequest.Headers["x-twilio-signature"];
-            validationUrl = validationUrl ?? (httpRequest.Headers["x-forwarded-proto"][0] ?? httpRequest.Protocol + "://" + httpRequest.Host + httpRequest.Path);
+            var urlString = validationUrl?.ToString();
+            if (string.IsNullOrEmpty(urlString))
```
[the remark this answers is not in the document]
Comment (gabog) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs]: Done (will go in the next commit, thanks)

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs [hunk 3]
```diff
@@ -120,12 +120,21 @@ public static Activity RequestToActivity(HttpRequest httpRequest, string validat
...
+            {
+                urlString = httpRequest.Headers["x-forwarded-proto"][0];
+                if (string.IsNullOrEmpty(urlString))
```
[the remark this answers is not in the document]
Comment (gabog) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs]: Done (will go in the next commit, thanks)

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

**Pull request merged**